src/sign_workflow/sign_artifacts.py

```diff
@@ -57,9 +59,9 @@ def __signer_class__(self, path: Path) -> Type[Any]:
             return SignArtifactsExistingArtifactFile
 
     @classmethod
-    def from_path(self, path: Path, components: List[str], artifact_type: str, signature_type: str, platform: str) -> Any:
+    def from_path(self, path: Path, components: List[str], artifact_type: str, signature_type: str, platform: str, overwrite: bool) -> Any:
```

Comment:
This should not take `overwrite`. The signer it returns is the same, whether we are trying to overwrite on signing or not.